### ISSUE TITLE
Selected text is wrapped in quotes when typing quotes

### ISSUE THREAD

**Issue by oliversalzburg** [microsoft/vscode#106740]
Issue Type: <b>Bug</b>

I'm pretty sure this isn't a bug, but by-design, but I can't find how to adjust the behavior (assuming you can).

When selecting a word and entering `"`, the word will be wrapped in quotes, instead of being replaced by a quote.

I searched the settings for "quote" and "wrap" but wasn't able to find a correlating setting.

VS Code version: Code 1.49.0 (e790b931385d72cf5669fcefc51cdf65990efa5d, 2020-09-10T13:22:08.892Z)
OS version: Windows_NT x64 10.0.19041
Remote OS version: Linux x64 4.19.104-microsoft-standard
Remote OS version: Linux x64 4.19.104-microsoft-standard
Remote OS version: Linux x64 4.19.104-microsoft-standard

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3672)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|63.91GB (45.46GB free)|
|Process Argv|--folder-uri=vscode-remote://wsl+Ubuntu/home/oliver/fairmanager/fairmanager/static2 --remote=wsl+Ubuntu --crash-reporter-id f825acfe-92fa-4526-a38a-f8815cf9f7ea|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.19.104-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3672)|
|Memory (System)|50.14GB (43.78GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.19.104-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3672)|
|Memory (System)|50.14GB (43.78GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.19.104-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3672)|
|Memory (System)|50.14GB (43.78GB free)|
|VM|0%|
</details><details><summary>Extensions (28)</summary>

Extension|Author (truncated)|Version
---|---|---
javascript-ejs-support|Dig|1.3.1
eslint-disable-snippets|drK|1.2.0
output-colorizer|IBM|0.1.2
vscode-docker|ms-|1.5.0
remote-containers|ms-|0.140.1
remote-ssh|ms-|0.55.0
remote-ssh-edit|ms-|0.55.0
remote-wsl|ms-|0.44.5
vscode-remote-extensionpack|ms-|0.20.0
vscode-icons|vsc|10.2.0
better-toml|bun|0.3.2
vscode-eslint|dba|2.1.8
eslint-disable-snippets|drK|1.2.0
gitlens|eam|10.2.2
EditorConfig|Edi|0.15.1
prettier-vscode|esb|5.6.0
vs-code-xml-format|fab|0.1.5
generateuuid|Fir|0.0.5
shell-format|fox|7.0.1
terraform|has|2.2.3
vscode-esformatter|how|1.6.0
code-beautifier|mic|2.3.3
vscode-docker|ms-|1.5.0
js-debug-nightly|ms-|2020.9.1417
vscode-paste-image|mus|1.0.4
vscode-xml|red|0.13.0
vscode-yaml|red|0.10.1
lit-plugin|run|1.2.1

(2 theme extensions excluded)

</details>

**Comment by IllusionMH** [comment]
This one is controlled by `editor.autoSurround` but I agree that it's quick to find even if this option is first result when searched by `selection`.

Probably it make sense to improve description.

Just in case you can do language specific settings like this:
```json
"[json]": {
    "editor.autoSurround": "never"
}

**Comment by oliversalzburg** [comment]
Oh, excellent. Thanks for pointing that out.

Not sure if that helps in finding a way to improve the discovery of the setting, but I noticed "quotes" is one of the possible values for `editor.autoSurround`. Not sure if it would make sense to also crawl the possible values of settings when filtering them down to a search phrase.

I found the `editor.autoClosingQuotes` through my search. That setting also felt very related to what I was looking for.

It feels a bit inconsistent that there a distinct settings to auto-close quotes/brackets. But for the surround it's a mutually exclusive choice.
